### PR TITLE
DATAREDIS 852 - Various NullPointerException using Lettuce and Range.Bound.unbounded()

### PR DESCRIPTION
**Pull request by mmanciop**
- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] There is a ticket in the bug tracker for the project in our [JIRA](https://jira.spring.io/browse/DATAREDIS).
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@mmanciop Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-data-redis&pullRequestId=353)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-data-redis&pullRequestId=353) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by mmanciop** [comment]
I am making separate changes for the different steps for ease of review. If this PR is accepted, please consider squashing the changes.

File: src/main/java/org/springframework/data/redis/connection/lettuce/LettuceReactiveZSetCommands.java [hunk 1]
```diff
@@ -189,28 +191,28 @@
 				if (command.isWithScores()) {
 
 					result = cmd
-							.zrangeWithScores(command.getKey(), command.getRange().getLowerBound().getValue().orElse(null),
-									command.getRange().getUpperBound().getValue().orElse(null))
+							.zrangeWithScores(command.getKey(), command.getRange().getLowerBound().getValue().orElse(Long.MIN_VALUE),
```
[the remark this answers is not in the document]
Comment (mmanciop) [reply src/main/java/org/springframework/data/redis/connection/lettuce/LettuceReactiveZSetCommands.java]: I am actually not 100% sure about this one. The range is based on scores, and scores can be negative, right?

[the remark this answers is not in the document]
Comment (mp911de) [reply src/main/java/org/springframework/data/redis/connection/lettuce/LettuceReactiveZSetCommands.java]: This one is https://redis.io/commands/zrange. You're mentioning https://redis.io/commands/zrangebyscore where min/max are based on scores and use `-inf`/`+inf` to denote min and max.

**Comment by mmanciop** [comment]
@mp911de I think I have addressed your remarks with the last push :-)

File: src/main/java/org/springframework/data/redis/connection/lettuce/LettuceReactiveZSetCommands.java [hunk 2]
```diff
@@ -189,28 +190,28 @@
...
+									command.getRange().getUpperBound().getValue().orElse(Long.MAX_VALUE))
 							.map(value -> (Tuple) new DefaultTuple(ByteUtils.getBytes(value), Double.NaN));
 				}
 			} else {
 				if (command.isWithScores()) {
 
 					result = cmd
-							.zrevrangeWithScores(command.getKey(), command.getRange().getLowerBound().getValue().orElse(null),
-									command.getRange().getUpperBound().getValue().orElse(null))
+							.zrevrangeWithScores(command.getKey(), command.getRange().getLowerBound().getValue().orElse(0L),
```
Comment (mmanciop) [inline src/main/java/org/springframework/data/redis/connection/lettuce/LettuceReactiveZSetCommands.java]: I do think that the -inf at https://redis.io/commands/zrevrangebyscore should be translated to Long.MIN_VALUE here.

File: src/main/java/org/springframework/data/redis/connection/lettuce/LettuceReactiveZSetCommands.java [hunk 3]
```diff
@@ -189,28 +190,28 @@
 				if (command.isWithScores()) {
 
 					result = cmd
-							.zrangeWithScores(command.getKey(), command.getRange().getLowerBound().getValue().orElse(null),
-									command.getRange().getUpperBound().getValue().orElse(null))
+							.zrangeWithScores(command.getKey(), command.getRange().getLowerBound().getValue().orElse(0L),
```
Comment (mmanciop) [inline src/main/java/org/springframework/data/redis/connection/lettuce/LettuceReactiveZSetCommands.java]: I do think that the -inf at https://redis.io/commands/zrangebyscore should be translated to Long.MIN_VALUE here.

File: src/main/java/org/springframework/data/redis/connection/lettuce/LettuceReactiveZSetCommands.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (mmanciop) [reply src/main/java/org/springframework/data/redis/connection/lettuce/LettuceReactiveZSetCommands.java]: Oh, OK then.

**Comment by mmanciop** [comment]
Retest this please

**Comment by mmanciop** [comment]
Looks like it timed out before running the ZSet and String tests.

**Comment by mp911de** [comment]
That's merged, polished, and backported now.

**Pull request closed without merging**